### PR TITLE
D15 2 more perf

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

**Comment by slluis** [comment]
Reviewers, I need more eyes on this.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionData.cs [hunk 1]
```diff
@@ -197,6 +197,11 @@ public virtual bool IsOverload (CompletionData other)
 			return DisplayText == other.DisplayText;
 		}
 
+		public override int GetHashCode ()
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionData.cs]: Why add this? there is no Equals() override.

Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionData.cs]: Custom IEqualityComparer which uses IsOverload as Equals

Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionData.cs]: I could probably get rid of this and just make the GetHashCode<T> return it.

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.Features/SemanticHighlighting/SemanticHighlightingVisitor.cs [hunk 2]
```diff
@@ -537,13 +537,9 @@ public override void VisitIdentifierName(Microsoft.CodeAnalysis.CSharp.Syntax.Id
...
-					symbol = semanticModel.GetSymbolInfo (node, cancellationToken).Symbol;
-					// var sym = vds.Variables[0].Initializer != null ? vds.Variables[0].Initializer.Value as LiteralExpressionSyntax : null;
-					if (symbol == null || symbol.Name != "var") {
-						Colorize(node.Span, varKeywordTypeColor);
-						return;
-					}
+				if (vds != null && vds.Variables.Count == 1) {
+					Colorize(node.Span, varKeywordTypeColor);
+					return;
 				}
```
Comment (DavidKarlas) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Features/SemanticHighlighting/SemanticHighlightingVisitor.cs]: @Therzok can you explain a bit why we don't need semantic confirmation for `VariableDeclarationSyntax`, but we need it on line 533 for `ForEachStatementSyntax`?

Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Features/SemanticHighlighting/SemanticHighlightingVisitor.cs]: That wasn't hit a lot, I just fixed whatever showed up via profiling. I guess that isn't needed either.

Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Features/SemanticHighlighting/SemanticHighlightingVisitor.cs]: On the other hand, that does not check whether the symbol is null or whether its name is var. Probably, we don't need to check the symbol there too.

**Pull request closed without merging**